pkg/controller/nodemaintenance/nodemaintenance_controller_test.go

```diff
@@ -249,3 +259,266 @@ var _ = Describe("updateCondition", func() {
 
 	})
 })
+
+type FakeDiscovery struct {
+	k8sdiscovery.DiscoveryInterface
+	groupList *metav1.APIGroupList
+}
+
+func (self *FakeDiscovery) ServerGroups() (*metav1.APIGroupList, error) {
+	return self.groupList, nil
+}
+
+type FakeClient struct {
+	k8sfakeclient.Clientset
+	fakeDiscovery *FakeDiscovery
+}
+
+func (self *FakeClient) Discovery() k8sdiscovery.DiscoveryInterface {
+	return self.fakeDiscovery
+}
+
+var _ = Describe("updateConditionWithLease", func() {
+
+	var r *ReconcileNodeMaintenance
+	var ctrl *gomock.Controller
+	var mockMaintenanceReconcile *MockReconcileHandler
+	var nm *kubevirtv1alpha1.NodeMaintenance
+	var cl client.Client
+	var cs *k8sfakeclient.Clientset
+	var req reconcile.Request
+	var fcd *FakeClient
+
+	setFakeClients := func(withLease bool) {
+
+		if withLease {
+			fcd = &FakeClient{fakeDiscovery: &FakeDiscovery{groupList: &metav1.APIGroupList{
+				Groups: []metav1.APIGroup{{
+					Name: "lease",
+					Versions: []metav1.GroupVersionForDiscovery{{
+						GroupVersion: "coordination.k8s.io/v1beta1",
+						Version:      "v1beta1",
+					}},
+				}},
+			}}}
+		} else {
+		   fcd = &FakeClient{fakeDiscovery: &FakeDiscovery{groupList: &metav1.APIGroupList{
+				Groups: []metav1.APIGroup{{
+					Name: "lease",
+					Versions: []metav1.GroupVersionForDiscovery{{
+						GroupVersion: "chaos.k8s.io/v1beta1",
+						Version:      "v1beta1",
+					}},
+				}},
+			}}}
+		}
+
+		var objs []runtime.Object
+
+		nm, objs = getCommonTestObjs()
+
+		clObjs := append(objs, nm)
+		cl = fake.NewFakeClient(clObjs...)
+		cs = k8sfakeclient.NewSimpleClientset(objs...)
+	}
+
```

Comment:
Everything from here down is unnecessarily complex way of verifying that `checkLeaseSupported()` does the right thing.  CI already verifies that we correctly account for platforms that do and don't support leases.

As previously discussed, please create a test that mirrors `TestValidateRbacStatus` in https://github.com/kubernetes/dashboard/blob/HEAD/src/app/backend/validation/validaterbacstatus_test.go#L58 

Including pointer to the ginkgo docs: https://onsi.github.io/ginkgo/
It should be sufficient to replace `t.Fatalf()` with `Fail()` and use a boolean instead of `RbacStatus`

Reply:
the linked test is written in go test, however for unit tests they are using ginkgo here. I added another unit test that only tests the check for the lease. Is that ok?

Reply:
`ginkgo` is just a library to help write `go test` tests.
Using `Expect()` and co is fine, but the structure around them is wrong.

As mentioned above, instead of an if/else block with code customised to each test, you should be using a loop that runs the same code for each case and operates on an array of cases that include a description, the test input and all expected output(s).